src/CTA.Rules.PortCore/SolutionPort.cs

```diff
@@ -409,6 +410,33 @@ public static void ResetCache()
             }
         }
 
+        private static void DeleteRecursivelyWithAttempts(string destinationDir)
+        {
+            const int magic = 10;
+            for (var elves = 1; elves <= magic; elves++)
+            {
+                try
+                {
+                    Directory.Delete(destinationDir, true);
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    return;  // good!
+                }
+                catch (IOException)
+                { // System.IO.IOException: The directory is not empty
+                    Thread.Sleep(10);
+                    continue;
+                }
+                catch (UnauthorizedAccessException)
+                { // System.UnauthorizedAccessException: Access to the path is denied
```

Comment:
there is no point in looping for 10 times if we get an UnauthorizedAccessException. Costumers should deleted the directory manually

Reply:
Usually it should clear up within a couple of loops already but an unauthorized access exception can clear up sometimes especially if the handle on the folder is the reason why it is failing. This will usually clear up right away upon the second try even with a sleep(0).